cpp/src/svm/svc_impl.cuh

```diff
@@ -355,48 +354,19 @@ void svcPredictSparse(const raft::handle_t& handle,
 template <typename math_t>
 void svmFreeBuffers(const raft::handle_t& handle, SvmModel<math_t>& m)
 {
-  cudaStream_t stream                      = handle.get_stream();
-  rmm::device_async_resource_ref rmm_alloc = rmm::mr::get_current_device_resource();
-  if (m.dual_coefs)
-    rmm_alloc.deallocate_async(
-      m.dual_coefs, m.n_support * sizeof(math_t), rmm::CUDA_ALLOCATION_ALIGNMENT, stream);
-  if (m.support_idx)
-    rmm_alloc.deallocate_async(
-      m.support_idx, m.n_support * sizeof(int), rmm::CUDA_ALLOCATION_ALIGNMENT, stream);
-  if (m.support_matrix.indptr) {
-    rmm_alloc.deallocate_async(m.support_matrix.indptr,
-                               (m.n_support + 1) * sizeof(int),
-                               rmm::CUDA_ALLOCATION_ALIGNMENT,
-                               stream);
-    m.support_matrix.indptr = nullptr;
-  }
-  if (m.support_matrix.indices) {
-    rmm_alloc.deallocate_async(m.support_matrix.indices,
-                               m.support_matrix.nnz * sizeof(int),
-                               rmm::CUDA_ALLOCATION_ALIGNMENT,
-                               stream);
-    m.support_matrix.indices = nullptr;
-  }
-  if (m.support_matrix.data) {
-    if (m.support_matrix.nnz == -1) {
-      rmm_alloc.deallocate_async(m.support_matrix.data,
-                                 m.n_support * m.n_cols * sizeof(math_t),
-                                 rmm::CUDA_ALLOCATION_ALIGNMENT,
-                                 stream);
-    } else {
-      rmm_alloc.deallocate_async(m.support_matrix.data,
-                                 m.support_matrix.nnz * sizeof(math_t),
-                                 rmm::CUDA_ALLOCATION_ALIGNMENT,
-                                 stream);
-    }
-  }
+  cudaStream_t stream = handle.get_stream();
+
+  // Note that the underlying allocations are not *freed* but rather reset
```

Comment:
I see that this is the case with `resize()`. Why not `deallocate_async()` instead to free memory?